### ISSUE TITLE
no implementation for removing and adding a bot admin

### ISSUE THREAD

**Issue by parmonov98** [badfarm/zanzara#72]
```
[nodemon] starting `php start.php`
Zanzara is listening...
/home/dev2-pc/Documents/php/group-counter/vendor/badfarm/zanzara/src/Zanzara/Context.php:74:
class Zanzara\Telegram\Type\Update#364 (18) {
  private $update_id =>
  int(757493498)
  private $message =>
  NULL
  private $edited_message =>
  NULL
  private $channel_post =>
  NULL
  private $edited_channel_post =>
  NULL
  private $inline_query =>
  NULL
  private $chosen_inline_result =>
  NULL
  private $callback_query =>
  NULL
  private $shipping_query =>
  NULL
  private $pre_checkout_query =>
  NULL
  private $poll =>
  NULL
  private $poll_answer =>
  NULL
  private $updateType =>
  NULL
  private $effectiveUser =>
  NULL
  private $effectiveChat =>
  NULL
  private $my_chat_member =>
  class Zanzara\Telegram\Type\ChatMember#121 (28) {
    private $user =>
    NULL
    private $status =>
    NULL
    private $custom_title =>
    NULL
    private $is_anonymous =>
    NULL
    private $until_date =>
    NULL
    private $can_be_edited =>
    NULL
    private $can_post_messages =>
    NULL
    private $can_edit_messages =>
    NULL
    private $can_delete_messages =>
    NULL
    private $can_restrict_members =>
    NULL
    private $can_promote_members =>
    NULL
    private $can_change_info =>
    NULL
    private $can_invite_users =>
    NULL
    private $can_pin_messages =>
    NULL
    private $is_member =>
    NULL
    private $can_send_messages =>
    NULL
    private $can_send_media_messages =>
    NULL
    private $can_send_polls =>
    NULL
    private $can_send_other_messages =>
    NULL
    private $can_add_web_page_previews =>
    NULL
    private $chat =>
    class Zanzara\Telegram\Type\Chat#99 (20) {
      private $id =>
      int(-1001673450624)
      private $type =>
      string(10) "supergroup"
      private $title =>
      string(14) "MyGroupCleaner"
      private $username =>
      string(14) "MyGroupCleaner"
      private $first_name =>
      NULL
      private $last_name =>
      NULL
      private $photo =>
      NULL
      private $bio =>
      NULL
      private $has_private_forwards =>
      NULL
      private $description =>
      NULL
      private $invite_link =>
      NULL
      private $pinned_message =>
      NULL
      private $permissions =>
      NULL
      private $slow_mode_delay =>
      NULL
      private $message_auto_delete_time =>
      NULL
      private $has_protected_content =>
      NULL
      private $sticker_set_name =>
      NULL
      private $can_set_sticker_set =>
      NULL
      private $linked_chat_id =>
      NULL
      private $location =>
      NULL
    }
    private $from =>
    class Zanzara\Telegram\Type\User#403 (9) {
      private $id =>
      int(94665561)
      private $is_bot =>
      bool(false)
      private $first_name =>
      string(5) "Murod"
      private $last_name =>
      string(8) "Parmonov"
      private $username =>
      string(10) "parmonov98"
      private $language_code =>
      NULL
      private $can_join_groups =>
      NULL
      private $can_read_all_group_messages =>
      NULL
      private $supports_inline_queries =>
      NULL
    }
    private $date =>
    int(1659675188)
    private $old_chat_member =>
    class Zanzara\Telegram\Type\ChatMember#286 (28) {
      private $user =>
      class Zanzara\Telegram\Type\User#268 (9) {
        ...
      }
      private $status =>
      string(13) "administrator"
      private $custom_title =>
      NULL
      private $is_anonymous =>
      bool(false)
      private $until_date =>
      NULL
      private $can_be_edited =>
      bool(false)
      private $can_post_messages =>
      NULL
      private $can_edit_messages =>
      NULL
      private $can_delete_messages =>
      bool(true)
      private $can_restrict_members =>
      bool(true)
      private $can_promote_members =>
      bool(false)
      private $can_change_info =>
      bool(true)
      private $can_invite_users =>
      bool(true)
      private $can_pin_messages =>
      bool(true)
      private $is_member =>
      NULL
      private $can_send_messages =>
      NULL
      private $can_send_media_messages =>
      NULL
      private $can_send_polls =>
      NULL
      private $can_send_other_messages =>
      NULL
      private $can_add_web_page_previews =>
      NULL
      private $chat =>
      NULL
      private $from =>
      NULL
      private $date =>
      NULL
      private $old_chat_member =>
      NULL
      private $new_chat_member =>
      NULL
      private $invite_link =>
      NULL
      private $can_manage_voice_chats =>
      bool(true)
      private $can_manage_chat =>
      bool(true)
    }
    private $new_chat_member =>
    class Zanzara\Telegram\Type\ChatMember#329 (28) {
      private $user =>
      class Zanzara\Telegram\Type\User#270 (9) {
        ...
      }
      private $status =>
      string(6) "member"
      private $custom_title =>
      NULL
      private $is_anonymous =>
      NULL
      private $until_date =>
      NULL
      private $can_be_edited =>
      NULL
      private $can_post_messages =>
      NULL
      private $can_edit_messages =>
      NULL
      private $can_delete_messages =>
      NULL
      private $can_restrict_members =>
      NULL
      private $can_promote_members =>
      NULL
      private $can_change_info =>
      NULL
      private $can_invite_users =>
      NULL
      private $can_pin_messages =>
      NULL
      private $is_member =>
      NULL
      private $can_send_messages =>
      NULL
      private $can_send_media_messages =>
      NULL
      private $can_send_polls =>
      NULL
      private $can_send_other_messages =>
      NULL
      private $can_add_web_page_previews =>
      NULL
      private $chat =>
      NULL
      private $from =>
      NULL
      private $date =>
      NULL
      private $old_chat_member =>
      NULL
      private $new_chat_member =>
      NULL
      private $invite_link =>
      NULL
      private $can_manage_voice_chats =>
      NULL
      private $can_manage_chat =>
      NULL
    }
    private $invite_link =>
    NULL
    private $can_manage_voice_chats =>
    NULL
    private $can_manage_chat =>
    NULL
  }
  private $chat_member =>
  NULL
  private $chat_join_request =>
  NULL
}
/home/dev2-pc/Documents/php/group-counter/vendor/badfarm/zanzara/src/Zanzara/Context.php:74:
class Zanzara\Telegram\Type\Update#364 (18) {
  private $update_id =>
  int(757493498)
  private $message =>
  NULL
  private $edited_message =>
  NULL
  private $channel_post =>
  NULL
  private $edited_channel_post =>
  NULL
  private $inline_query =>
  NULL
  private $chosen_inline_result =>
  NULL
  private $callback_query =>
  NULL
  private $shipping_query =>
  NULL
  private $pre_checkout_query =>
  NULL
  private $poll =>
  NULL
  private $poll_answer =>
  NULL
  private $updateType =>
  NULL
  private $effectiveUser =>
  NULL
  private $effectiveChat =>
  NULL
  private $my_chat_member =>
  class Zanzara\Telegram\Type\ChatMember#121 (28) {
    private $user =>
    NULL
    private $status =>
    NULL
    private $custom_title =>
    NULL
    private $is_anonymous =>
    NULL
    private $until_date =>
    NULL
    private $can_be_edited =>
    NULL
    private $can_post_messages =>
    NULL
    private $can_edit_messages =>
    NULL
    private $can_delete_messages =>
    NULL
    private $can_restrict_members =>
    NULL
    private $can_promote_members =>
    NULL
    private $can_change_info =>
    NULL
    private $can_invite_users =>
    NULL
    private $can_pin_messages =>
    NULL
    private $is_member =>
    NULL
    private $can_send_messages =>
    NULL
    private $can_send_media_messages =>
    NULL
    private $can_send_polls =>
    NULL
    private $can_send_other_messages =>
    NULL
    private $can_add_web_page_previews =>
    NULL
    private $chat =>
    class Zanzara\Telegram\Type\Chat#99 (20) {
      private $id =>
      int(-1001673450624)
      private $type =>
      string(10) "supergroup"
      private $title =>
      string(14) "MyGroupCleaner"
      private $username =>
      string(14) "MyGroupCleaner"
      private $first_name =>
      NULL
      private $last_name =>
      NULL
      private $photo =>
      NULL
      private $bio =>
      NULL
      private $has_private_forwards =>
      NULL
      private $description =>
      NULL
      private $invite_link =>
      NULL
      private $pinned_message =>
      NULL
      private $permissions =>
      NULL
      private $slow_mode_delay =>
      NULL
      private $message_auto_delete_time =>
      NULL
      private $has_protected_content =>
      NULL
      private $sticker_set_name =>
      NULL
      private $can_set_sticker_set =>
      NULL
      private $linked_chat_id =>
      NULL
      private $location =>
      NULL
    }
    private $from =>
    class Zanzara\Telegram\Type\User#403 (9) {
      private $id =>
      int(94665561)
      private $is_bot =>
      bool(false)
      private $first_name =>
      string(5) "Murod"
      private $last_name =>
      string(8) "Parmonov"
      private $username =>
      string(10) "parmonov98"
      private $language_code =>
      NULL
      private $can_join_groups =>
      NULL
      private $can_read_all_group_messages =>
      NULL
      private $supports_inline_queries =>
      NULL
    }
    private $date =>
    int(1659675188)
    private $old_chat_member =>
    class Zanzara\Telegram\Type\ChatMember#286 (28) {
      private $user =>
      class Zanzara\Telegram\Type\User#268 (9) {
        ...
      }
      private $status =>
      string(13) "administrator"
      private $custom_title =>
      NULL
      private $is_anonymous =>
      bool(false)
      private $until_date =>
      NULL
      private $can_be_edited =>
      bool(false)
      private $can_post_messages =>
      NULL
      private $can_edit_messages =>
      NULL
      private $can_delete_messages =>
      bool(true)
      private $can_restrict_members =>
      bool(true)
      private $can_promote_members =>
      bool(false)
      private $can_change_info =>
      bool(true)
      private $can_invite_users =>
      bool(true)
      private $can_pin_messages =>
      bool(true)
      private $is_member =>
      NULL
      private $can_send_messages =>
      NULL
      private $can_send_media_messages =>
      NULL
      private $can_send_polls =>
      NULL
      private $can_send_other_messages =>
      NULL
      private $can_add_web_page_previews =>
      NULL
      private $chat =>
      NULL
      private $from =>
      NULL
      private $date =>
      NULL
      private $old_chat_member =>
      NULL
      private $new_chat_member =>
      NULL
      private $invite_link =>
      NULL
      private $can_manage_voice_chats =>
      bool(true)
      private $can_manage_chat =>
      bool(true)
    }
    private $new_chat_member =>
    class Zanzara\Telegram\Type\ChatMember#329 (28) {
      private $user =>
      class Zanzara\Telegram\Type\User#270 (9) {
        ...
      }
      private $status =>
      string(6) "member"
      private $custom_title =>
      NULL
      private $is_anonymous =>
      NULL
      private $until_date =>
      NULL
      private $can_be_edited =>
      NULL
      private $can_post_messages =>
      NULL
      private $can_edit_messages =>
      NULL
      private $can_delete_messages =>
      NULL
      private $can_restrict_members =>
      NULL
      private $can_promote_members =>
      NULL
      private $can_change_info =>
      NULL
      private $can_invite_users =>
      NULL
      private $can_pin_messages =>
      NULL
      private $is_member =>
      NULL
      private $can_send_messages =>
      NULL
      private $can_send_media_messages =>
      NULL
      private $can_send_polls =>
      NULL
      private $can_send_other_messages =>
      NULL
      private $can_add_web_page_previews =>
      NULL
      private $chat =>
      NULL
      private $from =>
      NULL
      private $date =>
      NULL
      private $old_chat_member =>
      NULL
      private $new_chat_member =>
      NULL
      private $invite_link =>
      NULL
      private $can_manage_voice_chats =>
      NULL
      private $can_manage_chat =>
      NULL
    }
    private $invite_link =>
    NULL
    private $can_manage_voice_chats =>
    NULL
    private $can_manage_chat =>
    NULL
  }
  private $chat_member =>
  NULL
  private $chat_join_request =>
  NULL
}
Failed to process Telegram update , reason: TypeError: Zanzara\Telegram\Type\Update::getUpdateType(): Return value must be of type string, null returned in /home/dev2-pc/Documents/php/group-counter/vendor/badfarm/zanzara/src/Zanzara/Telegram/Type/Update.php:347
Stack trace:
#0 /home/dev2-pc/Documents/php/group-counter/vendor/badfarm/zanzara/src/Zanzara/Listener/ListenerResolver.php(33): Zanzara\Telegram\Type\Update->getUpdateType()
#1 /home/dev2-pc/Documents/php/group-counter/vendor/badfarm/zanzara/src/Zanzara/UpdateMode/UpdateMode.php(88): Zanzara\Listener\ListenerResolver->resolveListeners()
#2 /home/dev2-pc/Documents/php/group-counter/vendor/badfarm/zanzara/src/Zanzara/UpdateMode/Polling.php(78): Zanzara\UpdateMode\UpdateMode->processUpdate()
#3 /home/dev2-pc/Documents/php/group-counter/vendor/react/promise/src/FulfilledPromise.php(28): Zanzara\UpdateMode\Polling->Zanzara\UpdateMode\{closure}()
#4 /home/dev2-pc/Documents/php/group-counter/vendor/react/promise/src/Promise.php(134): React\Promise\FulfilledPromise->then()
#5 /home/dev2-pc/Documents/php/group-counter/vendor/react/promise/src/Promise.php(168): React\Promise\Promise::React\Promise\{closure}()
#6 /home/dev2-pc/Documents/php/group-counter/vendor/react/promise/src/Promise.php(231): React\Promise\Promise->settle()
#7 /home/dev2-pc/Documents/php/group-counter/vendor/react/promise/src/FulfilledPromise.php(42): React\Promise\Promise::React\Promise\{closure}()
#8 /home/dev2-pc/Documents/php/group-counter/vendor/react/promise/src/Promise.php(135): React\Promise\FulfilledPromise->done()
#9 /home/dev2-pc/Documents/php/group-counter/vendor/react/promise/src/Promise.php(168): React\Promise\Promise::React\Promise\{closure}()
#10 /home/dev2-pc/Documents/php/group-counter/vendor/react/promise/src/Promise.php(231): React\Promise\Promise->settle()
#11 /home/dev2-pc/Documents/php/group-counter/vendor/react/promise/src/Deferred.php(36): React\Promise\Promise::React\Promise\{closure}()
#12 /home/dev2-pc/Documents/php/group-counter/vendor/react/http/src/Io/Transaction.php(90): React\Promise\Deferred->resolve()
#13 /home/dev2-pc/Documents/php/group-counter/vendor/react/promise/src/FulfilledPromise.php(28): React\Http\Io\Transaction->React\Http\Io\{closure}()
#14 /home/dev2-pc/Documents/php/group-counter/vendor/react/promise/src/Promise.php(134): React\Promise\FulfilledPromise->then()
#15 /home/dev2-pc/Documents/php/group-counter/vendor/react/promise/src/Promise.php(168): React\Promise\Promise::React\Promise\{closure}()
#16 /home/dev2-pc/Documents/php/group-counter/vendor/react/promise/src/Promise.php(231): React\Promise\Promise->settle()
#17 /home/dev2-pc/Documents/php/group-counter/vendor/react/promise/src/FulfilledPromise.php(42): React\Promise\Promise::React\Promise\{closure}()
#18 /home/dev2-pc/Documents/php/group-counter/vendor/react/promise/src/Promise.php(135): React\Promise\FulfilledPromise->done()
#19 /home/dev2-pc/Documents/php/group-counter/vendor/react/promise/src/Promise.php(168): React\Promise\Promise::React\Promise\{closure}()
#20 /home/dev2-pc/Documents/php/group-counter/vendor/react/promise/src/Promise.php(231): React\Promise\Promise->settle()
#21 /home/dev2-pc/Documents/php/group-counter/vendor/react/promise/src/FulfilledPromise.php(42): React\Promise\Promise::React\Promise\{closure}()
#22 /home/dev2-pc/Documents/php/group-counter/vendor/react/promise/src/Promise.php(66): React\Promise\FulfilledPromise->done()
#23 /home/dev2-pc/Documents/php/group-counter/vendor/react/promise/src/Promise.php(168): React\Promise\Promise::React\Promise\{closure}()
#24 /home/dev2-pc/Documents/php/group-counter/vendor/react/promise/src/Promise.php(231): React\Promise\Promise->settle()
#25 /home/dev2-pc/Documents/php/group-counter/vendor/react/promise/src/FulfilledPromise.php(42): React\Promise\Promise::React\Promise\{closure}()
#26 /home/dev2-pc/Documents/php/group-counter/vendor/react/promise/src/Promise.php(135): React\Promise\FulfilledPromise->done()
#27 /home/dev2-pc/Documents/php/group-counter/vendor/react/promise/src/Promise.php(168): React\Promise\Promise::React\Promise\{closure}()
#28 /home/dev2-pc/Documents/php/group-counter/vendor/react/promise/src/Promise.php(231): React\Promise\Promise->settle()
#29 /home/dev2-pc/Documents/php/group-counter/vendor/react/promise/src/FulfilledPromise.php(42): React\Promise\Promise::React\Promise\{closure}()
#30 /home/dev2-pc/Documents/php/group-counter/vendor/react/promise/src/Promise.php(135): React\Promise\FulfilledPromise->done()
#31 /home/dev2-pc/Documents/php/group-counter/vendor/react/promise/src/Promise.php(168): React\Promise\Promise::React\Promise\{closure}()
#32 /home/dev2-pc/Documents/php/group-counter/vendor/react/promise/src/Promise.php(231): React\Promise\Promise->settle()
#33 /home/dev2-pc/Documents/php/group-counter/vendor/react/promise-stream/src/functions.php(79): React\Promise\Promise::React\Promise\{closure}()
#34 /home/dev2-pc/Documents/php/group-counter/vendor/evenement/evenement/src/Evenement/EventEmitterTrait.php(123): React\Promise\Stream\{closure}()
#35 /home/dev2-pc/Documents/php/group-counter/vendor/react/http/src/Io/ReadableBodyStream.php(50): Evenement\EventEmitter->emit()
#36 /home/dev2-pc/Documents/php/group-counter/vendor/react/http/src/Io/ReadableBodyStream.php(151): React\Http\Io\ReadableBodyStream->close()
#37 /home/dev2-pc/Documents/php/group-counter/vendor/react/http/src/Io/ReadableBodyStream.php(33): React\Http\Io\ReadableBodyStream->handleEnd()
#38 /home/dev2-pc/Documents/php/group-counter/vendor/evenement/evenement/src/Evenement/EventEmitterTrait.php(123): React\Http\Io\ReadableBodyStream->React\Http\Io\{closure}()
#39 /home/dev2-pc/Documents/php/group-counter/vendor/react/http/src/Client/Request.php(163): Evenement\EventEmitter->emit()
#40 /home/dev2-pc/Documents/php/group-counter/vendor/evenement/evenement/src/Evenement/EventEmitterTrait.php(123): React\Http\Client\Request->handleData()
#41 /home/dev2-pc/Documents/php/group-counter/vendor/react/stream/src/Util.php(71): Evenement\EventEmitter->emit()
#42 /home/dev2-pc/Documents/php/group-counter/vendor/evenement/evenement/src/Evenement/EventEmitterTrait.php(123): React\Stream\Util::React\Stream\{closure}()
#43 /home/dev2-pc/Documents/php/group-counter/vendor/react/stream/src/DuplexResourceStream.php(196): Evenement\EventEmitter->emit()
#44 /home/dev2-pc/Documents/php/group-counter/vendor/react/event-loop/src/StreamSelectLoop.php(246): React\Stream\DuplexResourceStream->handleData()
#45 /home/dev2-pc/Documents/php/group-counter/vendor/react/event-loop/src/StreamSelectLoop.php(213): React\EventLoop\StreamSelectLoop->waitForStreamActivity()
#46 /home/dev2-pc/Documents/php/group-counter/vendor/badfarm/zanzara/src/Zanzara/Zanzara.php(90): React\EventLoop\StreamSelectLoop->run()
#47 /home/dev2-pc/Documents/php/group-counter/start.php(81): Zanzara\Zanzara->run()
#48 {main}

```

**Comment by awohsen** [comment]
The `getUpdateType(): Return value must be of type string, null returned`, refers to an **unsupported update type**. 

There is a temporary solution to get these updates and get rid of the error logs [here #46](https://github.com/badfarm/zanzara/issues/46).

**Comment by cheeghi** [comment]
Fix released in the 0.9.0 version.
